addon/components/polaris-button.js

```diff
@@ -322,107 +318,110 @@ export default Component.extend({
+    }
 
-      if (loading) {
-        classes.push('Polaris-Button--loading');
-      }
+    if (isDisabled) {
+      classes.push('Polaris-Button--disabled');
+    }
 
-      if (plain) {
-        classes.push('Polaris-Button--plain');
-      }
+    if (loading) {
+      classes.push('Polaris-Button--loading');
+    }
 
-      if (monochrome) {
-        classes.push('Polaris-Button--monochrome');
-      }
+    if (plain) {
+      classes.push('Polaris-Button--plain');
+    }
 
-      if (size && size !== DEFAULT_SIZE) {
-        size = SIZES[size] || null;
-        if (size) {
-          classes.push(`Polaris-Button--size${classify(size)}`);
-        }
-      }
+    if (monochrome) {
+      classes.push('Polaris-Button--monochrome');
+    }
 
-      if (fullWidth) {
-        classes.push('Polaris-Button--fullWidth');
+    if (size && size !== DEFAULT_SIZE) {
+      size = SIZES[size] || null;
+      if (size) {
+        classes.push(`Polaris-Button--size${classify(size)}`);
       }
+    }
 
-      if (isIconOnly) {
-        classes.push('Polaris-Button--iconOnly');
-      }
+    if (fullWidth) {
+      classes.push('Polaris-Button--fullWidth');
+    }
 
-      return classes.join(' ');
+    if (isIconOnly) {
+      classes.push('Polaris-Button--iconOnly');
     }
-  ).readOnly(),
 
-  type: computed('submit', function() {
-    return this.get('submit') === true ? 'submit' : 'button';
-  }).readOnly(),
+    return classes.join(' ');
+  }
+
+  @computed('submit')
+  get type() {
+    return this.submit === true ? 'submit' : 'button';
+  }
 
-  ariaExpandedValue: computed('ariaExpanded', function() {
-    let ariaExpanded = this.get('ariaExpanded');
+  @computed('ariaExpanded')
+  get ariaExpandedValue() {
+    let ariaExpanded = this.ariaExpanded;
     return isPresent(ariaExpanded) ? String(ariaExpanded) : null;
-  }).readOnly(),
+  }
 
-  ariaPressedValue: computed('ariaPressed', function() {
-    let ariaPressed = this.get('ariaPressed');
+  @computed('ariaPressed')
+  get ariaPressedValue() {
+    let ariaPressed = this.ariaPressed;
     return isPresent(ariaPressed) ? String(ariaPressed) : null;
-  }).readOnly(),
-
-  actions: {
-    /**
-     * Helper to invoke passed-in actions without passing the
-     * button element's event object as the first parameter.
-     */
-    invokeMouseAction(actionName) {
-      return this.get(actionName)();
-    },
-  },
-});
+  }
+
+  init() {
+    super.init(...arguments);
+
+    deprecate(
+      `[polaris-button] Passing 'externalClasses' is deprecated! Switch to angle bracket invocation and pass an HTML attribute instead`,
+      !this.externalClasses,
+      {
+        id: 'ember-polaris.polaris-button.externalClasses-arg',
+        until: '6.0.0',
+      }
+    );
+  }
+
+  /**
+   * Helper to invoke passed-in actions without passing the
+   * button element's event object as the first parameter.
+   */
+  @action
+  invokeMouseAction(actionName) {
```

Comment:
I've never realised that we don't send back the event. What if someone wants the event? (to prevent bubbling)